runtime/runtime.go

```diff
@@ -2145,3 +2184,207 @@ func NewBlockValue(block Block) interpreter.BlockValue {
 		Timestamp: timestampValue,
 	}
 }
+
+func (r *interpreterRuntime) newAccountKeysAddFunction(
+	addressValue interpreter.AddressValue,
+	runtimeInterface Interface,
+) interpreter.HostFunctionValue {
+	return interpreter.NewHostFunctionValue(
+		func(invocation interpreter.Invocation) trampoline.Trampoline {
+			publicKeyValue := invocation.Arguments[0].(*interpreter.CompositeValue)
+			if publicKeyValue.QualifiedIdentifier != sema.PublicKeyTypeName {
+				panic(fmt.Sprintf(
+					"add method requires the first argument to be an %s",
+					sema.PublicKeyType,
+				))
+			}
+
+			hashAlgo := NewHashAlgorithmFromValue(invocation.Arguments[1])
+
+			weight, ok := invocation.Arguments[2].(interpreter.UFix64Value)
+			if !ok {
+				panic(fmt.Sprintf(
+					"add requires the third argument to be an %s",
+					sema.UFix64Type{},
+				))
+			}
+
+			var err error
+			var accountKey *AccountKey
+			wrapPanic(func() {
+				publicKey := NewPublicKeyFromValue(publicKeyValue)
+				accountKey, err = runtimeInterface.AddAccountKey(addressValue.ToAddress(), publicKey, hashAlgo, weight.ToInt())
+			})
```

Comment:
Only perform the runtime interface call in `wrapPanic`, so that only potential panics by the runtime interface call are treated as an external error

```suggestion
			publicKey := NewPublicKeyFromValue(publicKeyValue)
			address := addressValue.ToAddress()
			rawWeight := weight.ToInt()

			var err error
			var accountKey *AccountKey
			wrapPanic(func() {
				accountKey, err = runtimeInterface.AddAccountKey(address, publicKey, hashAlgo, rawWeight)
			})
```